### PR TITLE
SOLR-15376 Adding setMaxFutureMs with Long type Overload method

### PR DESCRIPTION
**Pull request by bitnahian**
https://issues.apache.org/jira/browse/SOLR-15376

# Description

`CollectionAdminRequest.CreateTimeRoutedAlias.setMaxFutureMs` method only accepts an integer value as a parameter. The Java Integer.MAX_VALUE is `2147483647`. This value is too small to accommodate for any duration greater than ~ 25 days. 

# Solution

Create an Overloaded method for setMaxFutureMs that accepts a Long type and change the type of instance variable `maxFutureMs` to a Long type and safely convert the original method's Integer type to a Long type in the original function. 

# Tests

Attempted to run tests under org/apache/solr/update/processor/TimeRoutedAliasUpdateProcessorTest.java but tests are currently ignored. 
# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Comment by bitnahian** [comment]
@cpoerschke Thanks for the feedback. Good catch. Missed this.

**Comment by cpoerschke** [comment]
> Attempted to run tests under org/apache/solr/update/processor/TimeRoutedAliasUpdateProcessorTest.java but tests are currently ignored.

Looks like `-Ptests.filter=@AwaitsFix` needs to be added to run the tests e.g.

```
./gradlew test --tests org.apache.solr.update.processor.TimeRoutedAliasUpdateProcessorTest.testMaxFutureMs -Ptests.filter=@AwaitsFix
```

for the test I've added in the commit above.

What do you think? Feel free to revert or amend, I've left some TODO comments on how it might change to use the new 'Long' type functionality.

**Comment by bitnahian** [comment]
Hi @cpoerschke, I've completed the TO-DOs suggested in your commit. :)

**Comment by cpoerschke** [comment]
Thanks @bitnahian for completing the test TO-DO items!

I just added two commits to the branch:
* @gus-asf's suggestion to deprecate the Integer method variant
* a draft solr/CHANGES.txt entry, question: would you prefer your name (Nahian-Al Hasan) or your github handle (bitnahian) or something else in the attribution?

File: solr/core/src/test/org/apache/solr/update/processor/TimeRoutedAliasUpdateProcessorTest.java [hunk 1]
```diff
@@ -713,6 +716,45 @@ public void testParse() {
...
   }
 
+  @Test
+  public void testMaxFutureMs() throws Exception {
+    final Long maxFutureMs = 100L * 24L * 60L * 60L * 1000L; // 100 DAYS
+
+    ZonedDateTime a = ZonedDateTime.now(ZoneId.from(ZoneOffset.UTC));
+    ZonedDateTime b = a.plusMonths(2);
+    ZonedDateTime c = a.plusYears(1);
+    DateTimeFormatter SOLR_DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss'Z'").withZone(ZoneId.from(ZoneOffset.UTC));
```
Comment (cpoerschke) [inline solr/core/src/test/org/apache/solr/update/processor/TimeRoutedAliasUpdateProcessorTest.java:726]:
```suggestion
    DateTimeFormatter SOLR_DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.ROOT).withZone(ZoneId.from(ZoneOffset.UTC));
```

or similar is what the CI feedback is probably saying; `./gradlew :solr:core:forbiddenApisTest` to run it locally.

**Comment by bitnahian** [comment]
@cpoerschke  bitnahian is fine :D

**Comment by cpoerschke** [comment]
Hmm, running the full test suite locally via `./gradlew test` gives two reproducible failures:

```
gradlew :solr:core:test --tests "org.apache.solr.handler.admin.V2CollectionsAPIMappingTest.testCreateAliasAllProperties" -Ptests.jvms=4 -Ptests.jvmargs=-XX:TieredStopAtLevel=1 -Ptests.seed=D0AC04911DD2F9B9 -Ptests.file.encoding=US-ASCII

gradlew :solr:core:test --tests "org.apache.solr.cloud.CreateRoutedAliasTest.testV2" -Ptests.jvms=4 -Ptests.jvmargs=-XX:TieredStopAtLevel=1 -Ptests.seed=D0AC04911DD2F9B9 -Ptests.file.encoding=US-ASCII
```

**Comment by cpoerschke** [comment]
> Hmm, running the full test suite locally via `./gradlew test` gives two reproducible failures: ...

This is, I think, because the `@JsonProperty` annotations don't yet support `Long` type, separate #285 pull request and JIRA issue opened for that.

**Comment by cpoerschke** [comment]
> So I'll just wait on the other issue to get resolved then? Any TO-DOs before then?

No TO-DOs that I can think of, thanks for the ping! Once the other issue is resolved the latest `solr/main` branch will need merging into this pull request's branch here.

**Review by cpoerschke — APPROVED** [review]

**Pull request merged**